src/v/cluster/self_test_frontend.h

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+#include "cluster/errc.h"
+#include "cluster/fwd.h"
+#include "cluster/members_table.h"
+#include "cluster/self_test_rpc_service.h"
+#include "cluster/self_test_rpc_types.h"
+#include "config/node_config.h"
+#include "rpc/connection_cache.h"
+#include "rpc/fwd.h"
+
+namespace cluster {
+
+class self_test_exception : public std::runtime_error {
+public:
+    explicit self_test_exception(ss::sstring msg)
+      : std::runtime_error(std::move(msg)) {}
+};
+
+class self_test_frontend {
```

Comment:
nit: here might be a good place to add a high level overview of how that self-test infra comes together -- that requests are meant to be sent to the leader, and they RPC out to all nodes, etc. Also that it's not critical to be done on the leader only (or maybe it is so our members table is most up-to-date?)